src/type-layout.md

```diff
@@ -73,6 +73,12 @@ at least equal to the size and alignment of a pointer.
 > title="Dynamically Sized Types">DSTs</abbr> are currently twice the size of
 > the size of `usize` and have the same alignment.
 
+## `str` Layout
+
+Strings are laid out so that the `nth` character of the string is offset from the
+start of the string by `n * the size of a character` bytes. A string has a size of
+`size_of::<char>() * n` and the same alignment of `char`.
```

Comment:
```rust
#![feature(dbg_macro)]

fn main() {
    use std::mem::*;

    let foo: &'static str = "foo";

    dbg!(size_of_val(foo));
    dbg!(size_of_val(&'f') + size_of_val(&'o') + size_of_val(&'o'));

    dbg!(align_of::<char>());
    dbg!(align_of_val(&'f'));
    dbg!(align_of_val(foo));
}
```

==>

```
[src/main.rs:8] size_of_val(foo) = 3
[src/main.rs:9] size_of_val(&'f') + size_of_val(&'o') + size_of_val(&'o') = 12
[src/main.rs:11] align_of::<char>() = 4
[src/main.rs:12] align_of_val(&'f') = 4
[src/main.rs:13] align_of_val(foo) = 1
```

This phrasing doesn't seem accurate?

Reply:
I see my mistakes. Thank you!